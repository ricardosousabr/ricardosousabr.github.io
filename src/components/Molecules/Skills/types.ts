export interface skillsProps {
  skillItens: string[];
}
