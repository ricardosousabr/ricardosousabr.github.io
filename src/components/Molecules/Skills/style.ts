import styled from "styled-components";
import screens from "../../../tokens/screen";

export const SkillStyle = styled.div`
  display: flex;
  flex-direction: column;
  justify-content: space-between;
  align-items: center;
  height: 700px;

  @media (min-width: ${screens.large}) {
    height: 400px;
  }

`;

export const BoxText = styled.div`
  display: flex;
  justify-content: space-between;
  flex-direction: column;
  align-items: center;
  height: 100px;

`
export const BoxSkillSItems = styled.div`
  display: flex;
  justify-content: center;
  flex-wrap: wrap;
  height: 800px;
  width: 300px;
  justify-content: space-between;

  @media (min-width: ${screens.large}) {
    width: 640px;
    height: 300px;

  }
`

export const BoxSkill = styled.div`
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  width: 78px;
  height: 96px;

  @media (min-width: ${screens.large}) {

  }
`
