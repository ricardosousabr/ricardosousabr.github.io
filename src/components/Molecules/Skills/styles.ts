import styled from "styled-components";
import screens from "../../../tokens/screen";

export const Wrapper = styled.div`
  background: #111827;
  padding: 50px 0;

`

export const SkillStyle = styled.div`
  display: flex;
  flex-direction: column;
  justify-content: space-between;
  align-items: center;
  text-align: center;

`;

export const BoxText = styled.div`
  display: flex;
  justify-content: space-between;
  flex-direction: column;
  align-items: center;
  padding-bottom: 30px;

`

export const BoxTitle = styled.div`
  padding-bottom: 30px;

`

export const BoxSkillSItems = styled.div`
  display: flex;
  padding: 0 50px;
  flex-wrap: wrap;
  justify-content: space-between;

    @media (min-width: ${screens.small}) {
      padding: 0 0px;

    }

`

export const BoxSkill = styled.div`
  display: flex;
  flex-direction: column;
  padding: 20px 20px;
  align-items: center;
  width: 40px;
  height: 40px;


`
