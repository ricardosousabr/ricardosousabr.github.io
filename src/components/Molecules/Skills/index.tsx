import Image from "next/image";
import Text from "../../Atoms/Text";
import { SkillStyle, BoxText, BoxSkillSItems, BoxSkill } from "./style";
import { skillsProps, } from "./types";

export default function Skills({ skillItens }: skillsProps) {
  skillItens = [
    "Next.js",
    "React.js",
    "JavaScript",
    "Storybook",
    "Firebase",
    "Styled-components",
    "HTML",
    "CSS",
    "Jest",
    "SASS",
    "Cypress.io",
    "Bootstrap",
    "JQuery",
    "Prettier",
    "ESLint",
  ];

  return (
    <SkillStyle>
      <BoxText>
        <Text color="white" fontSize="20px">Skills</Text>
        <Text color="white" fontSize="16px">The skills, tools and technologies I am good at:</Text>
      </BoxText>
      <BoxSkillSItems>
        {
          skillItens.map((skillItens, index) => {
            return (
              <BoxSkill key={index}>
                <Image width={40} height={40} src={"/img/"+skillItens+".png"} alt="" />
                <Text color="white" fontSize="16px">{skillItens}</Text>
              </BoxSkill>
            )
          })
        }
     </BoxSkillSItems>
    </SkillStyle>
  )
}