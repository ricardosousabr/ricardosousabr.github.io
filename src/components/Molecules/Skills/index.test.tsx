import renderer from "react-test-renderer";
import Skills from "./index";
import locale from "../../../locales";

test("Link renders correctly", () => {
  const { skills } = locale['pt-br'];
  const tree = renderer.create(<Skills {...skills} />).toJSON();
  expect(tree).toMatchSnapshot();
});
