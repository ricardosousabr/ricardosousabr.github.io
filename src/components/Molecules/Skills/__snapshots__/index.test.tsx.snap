// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Link renders correctly 1`] = `
<div
  className="sc-dlniIP dUoVBO"
>
  <div
    className="sc-gtsqUy iTcSzc"
  >
    <div
      className="sc-hKFymg iESXDq"
    >
      <div
        className="sc-eCAqax cXfcfV"
      >
        <div
          className="sc-jSFipO evjPyf"
        >
          <p
            className="sc-bdnyFh iOsQJx"
            color="white"
            fontSize="20px"
          >
            Skills
          </p>
        </div>
        <div>
          <p
            className="sc-bdnyFh iOusbc"
            color="white"
            fontSize="16px"
          >
            The skills, tools and technologies I am good at:
          </p>
        </div>
      </div>
      <div
        className="sc-gKAaef cwJAGK"
      >
        <div
          className="sc-iCoHzw eqvRVO"
        >
          <img
            alt=""
            data-nimg="1"
            decoding="async"
            height={40}
            loading="lazy"
            onError={[Function]}
            onLoad={[Function]}
            src="/_next/image?url=%2Fimg%2FNext.js.png&w=96&q=75"
            srcSet="/_next/image?url=%2Fimg%2FNext.js.png&w=48&q=75 1x, /_next/image?url=%2Fimg%2FNext.js.png&w=96&q=75 2x"
            style={
              {
                "color": "transparent",
              }
            }
            width={40}
          />
          <p
            className="sc-bdnyFh iOusbc"
            color="white"
            fontSize="16px"
          >
            Next.js
          </p>
        </div>
        <div
          className="sc-iCoHzw eqvRVO"
        >
          <img
            alt=""
            data-nimg="1"
            decoding="async"
            height={40}
            loading="lazy"
            onError={[Function]}
            onLoad={[Function]}
            src="/_next/image?url=%2Fimg%2FReact.js.png&w=96&q=75"
            srcSet="/_next/image?url=%2Fimg%2FReact.js.png&w=48&q=75 1x, /_next/image?url=%2Fimg%2FReact.js.png&w=96&q=75 2x"
            style={
              {
                "color": "transparent",
              }
            }
            width={40}
          />
          <p
            className="sc-bdnyFh iOusbc"
            color="white"
            fontSize="16px"
          >
            React.js
          </p>
        </div>
        <div
          className="sc-iCoHzw eqvRVO"
        >
          <img
            alt=""
            data-nimg="1"
            decoding="async"
            height={40}
            loading="lazy"
            onError={[Function]}
            onLoad={[Function]}
            src="/_next/image?url=%2Fimg%2FJavaScript.png&w=96&q=75"
            srcSet="/_next/image?url=%2Fimg%2FJavaScript.png&w=48&q=75 1x, /_next/image?url=%2Fimg%2FJavaScript.png&w=96&q=75 2x"
            style={
              {
                "color": "transparent",
              }
            }
            width={40}
          />
          <p
            className="sc-bdnyFh iOusbc"
            color="white"
            fontSize="16px"
          >
            JavaScript
          </p>
        </div>
        <div
          className="sc-iCoHzw eqvRVO"
        >
          <img
            alt=""
            data-nimg="1"
            decoding="async"
            height={40}
            loading="lazy"
            onError={[Function]}
            onLoad={[Function]}
            src="/_next/image?url=%2Fimg%2FStorybook.png&w=96&q=75"
            srcSet="/_next/image?url=%2Fimg%2FStorybook.png&w=48&q=75 1x, /_next/image?url=%2Fimg%2FStorybook.png&w=96&q=75 2x"
            style={
              {
                "color": "transparent",
              }
            }
            width={40}
          />
          <p
            className="sc-bdnyFh iOusbc"
            color="white"
            fontSize="16px"
          >
            Storybook
          </p>
        </div>
        <div
          className="sc-iCoHzw eqvRVO"
        >
          <img
            alt=""
            data-nimg="1"
            decoding="async"
            height={40}
            loading="lazy"
            onError={[Function]}
            onLoad={[Function]}
            src="/_next/image?url=%2Fimg%2FFirebase.png&w=96&q=75"
            srcSet="/_next/image?url=%2Fimg%2FFirebase.png&w=48&q=75 1x, /_next/image?url=%2Fimg%2FFirebase.png&w=96&q=75 2x"
            style={
              {
                "color": "transparent",
              }
            }
            width={40}
          />
          <p
            className="sc-bdnyFh iOusbc"
            color="white"
            fontSize="16px"
          >
            Firebase
          </p>
        </div>
        <div
          className="sc-iCoHzw eqvRVO"
        >
          <img
            alt=""
            data-nimg="1"
            decoding="async"
            height={40}
            loading="lazy"
            onError={[Function]}
            onLoad={[Function]}
            src="/_next/image?url=%2Fimg%2FStyled-components.png&w=96&q=75"
            srcSet="/_next/image?url=%2Fimg%2FStyled-components.png&w=48&q=75 1x, /_next/image?url=%2Fimg%2FStyled-components.png&w=96&q=75 2x"
            style={
              {
                "color": "transparent",
              }
            }
            width={40}
          />
          <p
            className="sc-bdnyFh iOusbc"
            color="white"
            fontSize="16px"
          >
            Styled-components
          </p>
        </div>
        <div
          className="sc-iCoHzw eqvRVO"
        >
          <img
            alt=""
            data-nimg="1"
            decoding="async"
            height={40}
            loading="lazy"
            onError={[Function]}
            onLoad={[Function]}
            src="/_next/image?url=%2Fimg%2FHTML.png&w=96&q=75"
            srcSet="/_next/image?url=%2Fimg%2FHTML.png&w=48&q=75 1x, /_next/image?url=%2Fimg%2FHTML.png&w=96&q=75 2x"
            style={
              {
                "color": "transparent",
              }
            }
            width={40}
          />
          <p
            className="sc-bdnyFh iOusbc"
            color="white"
            fontSize="16px"
          >
            HTML
          </p>
        </div>
        <div
          className="sc-iCoHzw eqvRVO"
        >
          <img
            alt=""
            data-nimg="1"
            decoding="async"
            height={40}
            loading="lazy"
            onError={[Function]}
            onLoad={[Function]}
            src="/_next/image?url=%2Fimg%2FCSS.png&w=96&q=75"
            srcSet="/_next/image?url=%2Fimg%2FCSS.png&w=48&q=75 1x, /_next/image?url=%2Fimg%2FCSS.png&w=96&q=75 2x"
            style={
              {
                "color": "transparent",
              }
            }
            width={40}
          />
          <p
            className="sc-bdnyFh iOusbc"
            color="white"
            fontSize="16px"
          >
            CSS
          </p>
        </div>
        <div
          className="sc-iCoHzw eqvRVO"
        >
          <img
            alt=""
            data-nimg="1"
            decoding="async"
            height={40}
            loading="lazy"
            onError={[Function]}
            onLoad={[Function]}
            src="/_next/image?url=%2Fimg%2FJest.png&w=96&q=75"
            srcSet="/_next/image?url=%2Fimg%2FJest.png&w=48&q=75 1x, /_next/image?url=%2Fimg%2FJest.png&w=96&q=75 2x"
            style={
              {
                "color": "transparent",
              }
            }
            width={40}
          />
          <p
            className="sc-bdnyFh iOusbc"
            color="white"
            fontSize="16px"
          >
            Jest
          </p>
        </div>
        <div
          className="sc-iCoHzw eqvRVO"
        >
          <img
            alt=""
            data-nimg="1"
            decoding="async"
            height={40}
            loading="lazy"
            onError={[Function]}
            onLoad={[Function]}
            src="/_next/image?url=%2Fimg%2FSASS.png&w=96&q=75"
            srcSet="/_next/image?url=%2Fimg%2FSASS.png&w=48&q=75 1x, /_next/image?url=%2Fimg%2FSASS.png&w=96&q=75 2x"
            style={
              {
                "color": "transparent",
              }
            }
            width={40}
          />
          <p
            className="sc-bdnyFh iOusbc"
            color="white"
            fontSize="16px"
          >
            SASS
          </p>
        </div>
        <div
          className="sc-iCoHzw eqvRVO"
        >
          <img
            alt=""
            data-nimg="1"
            decoding="async"
            height={40}
            loading="lazy"
            onError={[Function]}
            onLoad={[Function]}
            src="/_next/image?url=%2Fimg%2FCypress.io.png&w=96&q=75"
            srcSet="/_next/image?url=%2Fimg%2FCypress.io.png&w=48&q=75 1x, /_next/image?url=%2Fimg%2FCypress.io.png&w=96&q=75 2x"
            style={
              {
                "color": "transparent",
              }
            }
            width={40}
          />
          <p
            className="sc-bdnyFh iOusbc"
            color="white"
            fontSize="16px"
          >
            Cypress.io
          </p>
        </div>
        <div
          className="sc-iCoHzw eqvRVO"
        >
          <img
            alt=""
            data-nimg="1"
            decoding="async"
            height={40}
            loading="lazy"
            onError={[Function]}
            onLoad={[Function]}
            src="/_next/image?url=%2Fimg%2FBootstrap.png&w=96&q=75"
            srcSet="/_next/image?url=%2Fimg%2FBootstrap.png&w=48&q=75 1x, /_next/image?url=%2Fimg%2FBootstrap.png&w=96&q=75 2x"
            style={
              {
                "color": "transparent",
              }
            }
            width={40}
          />
          <p
            className="sc-bdnyFh iOusbc"
            color="white"
            fontSize="16px"
          >
            Bootstrap
          </p>
        </div>
        <div
          className="sc-iCoHzw eqvRVO"
        >
          <img
            alt=""
            data-nimg="1"
            decoding="async"
            height={40}
            loading="lazy"
            onError={[Function]}
            onLoad={[Function]}
            src="/_next/image?url=%2Fimg%2FJQuery.png&w=96&q=75"
            srcSet="/_next/image?url=%2Fimg%2FJQuery.png&w=48&q=75 1x, /_next/image?url=%2Fimg%2FJQuery.png&w=96&q=75 2x"
            style={
              {
                "color": "transparent",
              }
            }
            width={40}
          />
          <p
            className="sc-bdnyFh iOusbc"
            color="white"
            fontSize="16px"
          >
            JQuery
          </p>
        </div>
        <div
          className="sc-iCoHzw eqvRVO"
        >
          <img
            alt=""
            data-nimg="1"
            decoding="async"
            height={40}
            loading="lazy"
            onError={[Function]}
            onLoad={[Function]}
            src="/_next/image?url=%2Fimg%2FPrettier.png&w=96&q=75"
            srcSet="/_next/image?url=%2Fimg%2FPrettier.png&w=48&q=75 1x, /_next/image?url=%2Fimg%2FPrettier.png&w=96&q=75 2x"
            style={
              {
                "color": "transparent",
              }
            }
            width={40}
          />
          <p
            className="sc-bdnyFh iOusbc"
            color="white"
            fontSize="16px"
          >
            Prettier
          </p>
        </div>
        <div
          className="sc-iCoHzw eqvRVO"
        >
          <img
            alt=""
            data-nimg="1"
            decoding="async"
            height={40}
            loading="lazy"
            onError={[Function]}
            onLoad={[Function]}
            src="/_next/image?url=%2Fimg%2FESLint.png&w=96&q=75"
            srcSet="/_next/image?url=%2Fimg%2FESLint.png&w=48&q=75 1x, /_next/image?url=%2Fimg%2FESLint.png&w=96&q=75 2x"
            style={
              {
                "color": "transparent",
              }
            }
            width={40}
          />
          <p
            className="sc-bdnyFh iOusbc"
            color="white"
            fontSize="16px"
          >
            ESLint
          </p>
        </div>
      </div>
    </div>
  </div>
</div>
`;
